
Synology library path is a formulaic but atypical.  Register an RPATH to avoid
requiring an additional LD_LIBRARY_PATH everywhere it's used.  Editing both the
build configure plus the upstream .ac in case it's regenerated

diff -cr Python-3.5.5/configure x/configure
*** configure	2018-06-16 07:30:43.000454610 +0000
--- configure	2018-06-16 08:34:50.801838237 +0000
***************
*** 5971,5977 ****
            ;;
      Linux*|GNU*|NetBSD*|FreeBSD*|DragonFly*|OpenBSD*)
  	  LDLIBRARY='libpython$(LDVERSION).so'
! 	  BLDLIBRARY='-L. -lpython$(LDVERSION)'
  	  RUNSHARED=LD_LIBRARY_PATH=`pwd`${LD_LIBRARY_PATH:+:${LD_LIBRARY_PATH}}
  	  INSTSONAME="$LDLIBRARY".$SOVERSION
  	  if test "$with_pydebug" != yes
--- 5971,5977 ----
            ;;
      Linux*|GNU*|NetBSD*|FreeBSD*|DragonFly*|OpenBSD*)
  	  LDLIBRARY='libpython$(LDVERSION).so'
! 	  BLDLIBRARY='-Wl,-rpath,$(LIBDIR) -L. -lpython$(LDVERSION)'
  	  RUNSHARED=LD_LIBRARY_PATH=`pwd`${LD_LIBRARY_PATH:+:${LD_LIBRARY_PATH}}
  	  INSTSONAME="$LDLIBRARY".$SOVERSION
  	  if test "$with_pydebug" != yes
diff -cr /home/bob/src/spksrc-allanc/spk/homeassistant/work/Python-3.5.5/configure.ac x/configure.ac
*** configure.ac	2018-06-16 07:30:38.048438143 +0000
--- configure.ac	2018-06-16 08:35:34.921999207 +0000
***************
*** 1127,1133 ****
            ;;
      Linux*|GNU*|NetBSD*|FreeBSD*|DragonFly*|OpenBSD*)
  	  LDLIBRARY='libpython$(LDVERSION).so'
! 	  BLDLIBRARY='-L. -lpython$(LDVERSION)'
  	  RUNSHARED=LD_LIBRARY_PATH=`pwd`${LD_LIBRARY_PATH:+:${LD_LIBRARY_PATH}}
  	  INSTSONAME="$LDLIBRARY".$SOVERSION
  	  if test "$with_pydebug" != yes
--- 1127,1133 ----
            ;;
      Linux*|GNU*|NetBSD*|FreeBSD*|DragonFly*|OpenBSD*)
  	  LDLIBRARY='libpython$(LDVERSION).so'
! 	  BLDLIBRARY='-Wl,-rpath,$(LIBDIR) -L. -lpython$(LDVERSION)'
  	  RUNSHARED=LD_LIBRARY_PATH=`pwd`${LD_LIBRARY_PATH:+:${LD_LIBRARY_PATH}}
  	  INSTSONAME="$LDLIBRARY".$SOVERSION
  	  if test "$with_pydebug" != yes
